analytics/analytics.go

```diff
@@ -202,6 +206,18 @@ func saveData(server string, data *EarthlyAnalytics) error {
 	return nil
 }
 
+// Count increases the global count of (subsystem, key) which then gets reported when CollectAnalytics is called.
+func Count(subsystem, key string) {
+	countsMutex.Lock()
+	defer countsMutex.Unlock()
+	m, ok := counts[subsystem]
+	if !ok {
+		m = map[string]int{}
+		counts[subsystem] = m
+	}
+	m[key]++
+}
+
```

Comment:
Why not `sync.Map`? https://golang.org/pkg/sync/#Map

Reply:
I wasn't familiar with sync.Map. I just watched [GopherCon 2017 - Lightning Talk: Bryan C Mills - An overview of sync.Map
](https://www.youtube.com/watch?v=C1EtfDnsdDs) which explains a bit about it. I'm not sure it'll be good for our use case since we're doing a lot of writes to the same key (i.e. incrementing the value of a specific key).

The go docs mention this:
```
The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys.
```

so I think it's best to stick with the current mutex+lock approach.

Furthermore, I'm unsure of sync.Map could (easily) handle being serialized to json.